src/controllers/AuthController.js

```diff
@@ -0,0 +1,74 @@
+/* eslint-disable no-underscore-dangle */
+/* eslint-disable camelcase */
+/* eslint-disable class-methods-use-this */
+import db from '../models';
+import JWTToken from '../utils/jwt';
+import { randomPw } from '../utils/helpers';
+import hash from '../utils/hash';
+import Responses from '../utils/Response';
+
+export const getTokenAfterSignIn = async (res, user) => {
+  try {
+    const userDetails = user.dataValues;
+    const token = await JWTToken.signToken(userDetails);
+    return Responses.handleSuccess(201, 'User logged in successfully', res, { token });
+  } catch (e) {
+    return Responses.handleError(500, e.message, res);
+  }
+};
+
+/** Auth Class */
+class AuthController {
+  /**
+   * facebookSignin
+   *
+   * @param {object} req - request object
+   * @param {object} res - response object
+   *
+   * @returns {object} - object containing a user object and token
+   * @memberof AuthController
+   *
+   */
+  async facebookSignin(req, res) {
+    // eslint-disable-next-line no-console
+    const { first_name, last_name, email } = req.user._json;
+    await db.user.findOrCreate({
+      where: { email },
+      defaults: {
+        firstName: first_name,
+        lastName: last_name,
+        email,
+        isVerified: true,
+        password: hash.generateSync(randomPw(8)),
+      }
```

Comment:
I think you may need to wrap this in a try - catch block to handle async errors.

Reply:
I used `catchErrors` function instead to keep our test coverage.